File: leetcode/binarySearch.cpp
#include<iostream>
#include<vector>

using namespace std;

int binarySearch(vector<int> vec, int number){
  auto beg = vec.begin(), end = vec.end();
  auto mid = vec.begin() + (end - beg) / 2;
  
  while(mid != end && *mid != number){
    if(number < *mid){
      end = mid;
    }
    else
      beg = mid + 1;
    mid = (beg + end) / 2;
  }
  return *mid;
}

int main(){
  vector<int> vec = {3,5,7,8,9,12,15};
  int number = 0;
  cout << " Enter number to search: " << endl;
  cin >> number;
  cout << binarySearch(vec,number) << endl;
  
  
}

File: leetcode/binarySearch.py
def binarySearch(vec, begin, end, number):
  if(len(vec) == 1 and vec[(begin + end) / 2] != number):
    return 'not found'
  if(vec[(begin + end) / 2] == number):
    return 'found'
  else:
    for i in range(begin, end):
      if vec[i] < number:
	return binarySearch(vec, (begin + end) / 2, end, number)
      elif vec[i] > number:
	return binarySearch(vec, begin, (begin + end) / 2, number)
  
def main():
  vec = [3,4,6,8,9,24,56,78]
  number = int(raw_input('Enter a number: '))
  print binarySearch(vec, 0, len(vec), number)
  
main()